database/database_test.go

```diff
@@ -234,3 +234,180 @@ func TestStatesWithStatus(t *testing.T) {
+		[]IneligibleDomainState{
+			{Name: "e.test", Policy: "bulk-1-year"},
+			{Name: "f.test", Policy: "bulk-1-year"},
+			{Name: "g.test", Policy: "bulk-18-week"},
+			{Name: "h.test", Policy: "bulk-18-week"},
+			{Name: "i.test", Policy: "bulk-1-year"},
+			{Name: "j.test", Policy: "bulk-18-week"},
+			{Name: "k.test", Policy: "bulk-1-year"},
+		},
+	},
+}
+
+// Test GetIneligibleDomainStates
+func TestGetIneligibleDomainStates(t *testing.T) {
+
+	resetDB()
+
+	// domainStates should be empty as domains are not added to database
+	domainStates, err := testDB.GetIneligibleDomainStates([]string{"a.test"})
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+	// add domains to the database
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+	for _, tt := range getAndDeleteTests {
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.wantStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+	}
+	// get domains from the database
+	for _, tr := range getAndDeleteTests {
+		domainStates, err = testDB.GetIneligibleDomainStates(tr.domainStates)
+		if err != nil {
+			t.Errorf("%s", err)
+		}
+
+		sort.Slice(domainStates, func(i, j int) bool { return domainStates[i].Name < domainStates[j].Name })
+		if len(domainStates) != len(tr.domainStates) {
+			t.Errorf("Incorrect count of states for test %s", tr.description)
+		}
+		for i, domainState := range domainStates {
+			if domainState.Name != (tr.domainStates[i]) {
+				t.Errorf("unexpected domain at position %d for test %s: %#v", i, tr.description, domainState)
+			}
+		}
+	}
+}
+
+// TestDeleteIneligibleDomainStates tests the DeleteIneligibleDomainStates function
+func TestDeleteIneligibleDomainStates(t *testing.T) {
+
+	resetDB()
+
+	// domainStates should be empty as domains are not added to database
+	domainStates, err := testDB.GetIneligibleDomainStates([]string{"a.test"})
+
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+	// add domains to the database
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+	for _, tt := range getAndDeleteTests {
+		err := testDB.SetIneligibleDomainStates(
+			tt.wantStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+	}
+	// delete domains from the database
+	for _, tr := range getAndDeleteTests {
+		err = testDB.DeleteIneligibleDomainStates(tr.domainStates)
+		if err != nil {
+			t.Errorf("%s", err)
+		}
+	}
+
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+}
```

Comment:
Some thoughts of other things to test:

- calls to the new functions with duplicate entries
- calls to get and delete for entries that don't exist

Reply:
I added a new function that tests for duplicate entries.  This would also test delete for entries that do not exist and get entries that do not exist is tested on line 462!